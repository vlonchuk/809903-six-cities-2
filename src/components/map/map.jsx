import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import leaflet from 'leaflet';

export default class Map extends PureComponent {
  constructor(props) {
    super(props);

    this._mapRef = React.createRef();
  }

  render() {
    return <section ref={this._mapRef} className="cities__map map" id="map">
    </section>;
  }

  componentDidMount() {
    const city = [52.38333, 4.9];
    const icon = leaflet.icon({
      iconUrl: `img/pin.svg`,
      iconSize: [30, 30]
    });

    const zoom = 12;
    if (this._mapRef.current) {
      const map = leaflet.map(this._mapRef.current, {
        center: city,
        zoom,
        zoomControl: false,
        marker: true
      });
      map.setView(city, zoom);

      leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);

      this.props.properties.forEach((el) => {
        leaflet
        .marker([el.coor.latitude, el.coor.longitude], {icon})
        .addTo(map);
      });
    }
  }
}

Map.propTypes = {
  properties: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    caption: PropTypes.string.isRequired,
    imgSrc: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    priceCurrency: PropTypes.string.isRequired,
    priceValue: PropTypes.number.isRequired,
    priceText: PropTypes.string.isRequired,
    coor: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
    }),
  })).isRequired,
};
