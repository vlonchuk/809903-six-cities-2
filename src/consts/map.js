export const ICON_URL = `/img/pin.svg`;
export const ICON_URL_ACTIVE = `/img/pin-active.svg`;
export const ICON_SIZE = 30;
export const TILE_LAYER = `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`;
export const ATTRIBUTION = `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`;
