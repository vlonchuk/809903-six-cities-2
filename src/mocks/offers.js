const offers = [
  {
    id: `prop-1`,
    city: `Amsterdam`,
    caption: `Beautiful & luxurious apartment at great location`,
    imgSrc: `img/apartment-01.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 120,
    priceText: `night`,
    rating: 50,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    },
  },
  {
    id: `prop-2`,
    city: `Amsterdam`,
    caption: `Wood and stone place`,
    imgSrc: `img/room.jpg`,
    type: `Private room`,
    priceCurrency: `€`,
    priceValue: 80,
    priceText: `night`,
    rating: 80,
    coor: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
  },
  {
    id: `prop-3`,
    city: `Amsterdam`,
    caption: `Canal View Prinsengracht`,
    imgSrc: `img/apartment-02.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 132,
    priceText: `night`,
    rating: 100,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
  },
  {
    id: `prop-4`,
    city: `Amsterdam`,
    caption: `Nice, cozy, warm big bed apartment`,
    imgSrc: `img/apartment-03.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 180,
    priceText: `night`,
    rating: 65,
    coor: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198
    },
  },
  {
    id: `prop-5`,
    city: `Paris`,
    caption: `Canal View Prinsengracht`,
    imgSrc: `img/apartment-02.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 132,
    priceText: `night`,
    rating: 90,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
  },
  {
    id: `prop-6`,
    city: `Paris`,
    caption: `Nice, cozy, warm big bed apartment`,
    imgSrc: `img/apartment-03.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 180,
    priceText: `night`,
    rating: 40,
    coor: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198
    },
  },
];

export const getPropertiesByCity = (city) => {
  return offers.filter((el) => el.city === city)
               .slice();
};

export default offers;
