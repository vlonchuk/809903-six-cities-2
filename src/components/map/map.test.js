import React from 'react';
import renderer from 'react-test-renderer';
import {Map} from './map';
import offers from './../../mocks/offers.js';

it(`Map correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<Map properties={offers} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
