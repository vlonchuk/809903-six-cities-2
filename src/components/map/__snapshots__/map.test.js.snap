// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map correctly renders after relaunch 1`] = `
<section
  className="cities__map map"
  id="map"
/>
`;
